private-templates-service/adaptivecards-templating-service/src/TemplateServiceClient.ts

```diff
@@ -401,11 +403,14 @@ export class TemplateServiceClient {
               continue;
             }
             else {
+              // Create new draft version
+              if (isShareable) instance.isShareable = isShareable;
               templateInstances.push(instance);
-              // Pushing existing deprecated version back
-              version = incrementVersion(existingTemplate);
-              templateInstance = setTemplateInstanceParam(templateInstance, templateData, templateState, isShareable, version);
-              templateInstances.push(templateInstance);
+              if (template || state) {
+                version = incrementVersion(existingTemplate);
+                templateInstance = setTemplateInstanceParam(templateInstance, templateData, templateState, isShareable, version);
+                templateInstances.push(templateInstance);
+              }
```

Comment:
in this **else** block code is repeated from the first **if** block. May be try to refactor it? Either put it both under one if statement with **or** condition or extract it to a method?

Reply:
Done, thanks!